### DISCUSSION TITLE
Steps to generate phosphosites_99.tsv file?

### DISCUSSION THREAD

**Discussion by seanso95** [vdemichev/DiaNN#1130] [Q&A] [answered]
Hi Vadim and all,

I am just wondering, is it possible to get a sense of what the filters/steps are that are used to generate the phosphosites_90 and phosphosites_99 files? I have recently run analysis on phosphoproteomic DIA data, with peptidoform scoring and MBR enabled (using a DIA-NN-generated in silico library).

I am interested in reprocessing the main parquet output and perhaps even using a different method to aggregate intensities (e.g. mean/median rather than Top 1) and am having difficulty digesting the file to match something similar to the phosphosites files.

Thank you,
Sean

**Comment by vdemichev** [comment, marked as answer]
Hi Sean,

In a bit more detail what's done to produce those matrices:

- both lib & global q-value filter at 0.5 (that is 50%)
- both lib & global peptidoform q-value filter at 0.5
- run-specific q-value filter 0.01
- run-specific PEP filter 0.01
- run-specific peptidoform q-value filter 0.01
- if using channels, run-specific channel q-value filter 0.01

Out of the remaining precursors, in a given run, those containing the site with localisaiton confidence >= 0.9 or 0.99 are chosen. Out of those, the one with the maximum quantity is chosen. The site quantity in that run is defined as equal to that quantity.

Best,
Vadim

**Reply by sooheon** [reply]
In the readme, you say:

```
When using MBR (or its imitation) and relying on the main .parquet report (recommended) instead of the quantitative matrices, use the following q-value filters:

Lib.Q.Value instead of Global.Q.Value
When applying a filter to Q.Value that is more stringent than the FDR threshold used to generate the DIA library (e.g. Q.Value < 0.001 filter), always apply the same filter to Lib.Q.Value
Lib.PG.Q.Value instead of Global.PG.Q.Value
Lib.Peptidoform.Q.Value instead of Global.Peptidoform.Q.Value, when using peptidoform scoring
```

This suggests using the lib value instead of the global value, but here you apply both filters? 

In my output, both is more stringent than just lib.

**Reply by vdemichev** [reply]
Yes, both filters are applied 'just in case' for extra confidence.

**Comment by lori-koch** [comment]
Hi, just to confirm, you are using the Site.Occupancy.Probabilities for each site and not the single PTM.Site.Confidence score for filtering to sites_90 or sites_99 ?

**Reply by vdemichev** [reply]
Correct

**Reply by lori-koch** [reply]
Thank you. For sites that can be assigned to either of 2 genes for example RPL23A;RPL23B, how is the Gene name listed in the sites file selected? In these cases I see either the second or first gene of the two chosen so don't understand how that's assigned? Thanks.

**Reply by lori-koch** [reply]
It's based on the first listed Systematic Id, I see now

**Comment by KR-15** [comment]
Hello there,

I also want to have more flexibility regarding site_report filtering and not necessarly only rely on the report.phosphosites_90.tsv or report.phosphosites_99.tsv output. I have run a phosphoproteomics experiment on 12 phospho-enriched samples using DIA-NN v2.6.1 and the recommended settings in the documentation.

@vdemichev Using R I've start to perform what you suggest and here is a recap that summarize everything:

1. Load the report.parquet and report.site_report.parquet file using arrow R package (`read_parquet()`)
2. Identify common columns between both matrices (**Run.Index, Run, Channel, Precursor.Id, Precursor.Lib.Index**). I know that in the documentation you precise the following: 

> The entries in the site report can be matched to the main report and annotated with the information from it, as each row of the site report corresponds to a single row in the main report, defined by the combination of the **run index**, **channel name** and **precursor library index**

but I have still used all the 5 common columns to join both matrices using `left_join(site_report, report, by = common_cols).`

3. Then I have start to filter this merged matrix using the above filtering that you explain to @seanso95:

> both lib & global q-value filter at 0.5 (that is 50%)
both lib & global peptidoform q-value filter at 0.5
run-specific q-value filter 0.01
run-specific PEP filter 0.01
run-specific peptidoform q-value filter 0.01
if using channels, run-specific channel q-value filter 0.01

Using all the filters, it results to the following code (I have followed the same order)

`
merged_report_filt <- merged_report %>% 
  filter(Lib.Q.Value <= 0.5 & Global.Q.Value <= 0.5) %>% 
  filter(Lib.Peptidoform.Q.Value <= 0.5 & Global.Peptidoform.Q.Value <= 0.5) %>% 
  filter(Q.Value <= 0.01) %>% 
  filter(PEP <= 0.01) %>% 
  filter(Peptidoform.Q.Value <= 0.05)

`
Note that no channel was used so the last filter was omit.

4. Then I need to use the **Site.Occupancy.Probabilities** to filter to 0.9 or 0.99 (or any other threshold)

5. But then what about the quantity that we need to choose ? I am a little blocked on that part. 
Do we need to use the **Fragment.Sum" quantities to get the quantitative values that are usually present in the report.phosphosites_90.tsv ? Or any other column ?

Here are the columns that I have in my filtered matrix:

> [1] "Averagine"                    "Best.Fr.Mz"                   "Best.Fr.Mz.Delta"             "Channel"                     
>  [5] "Channel.Evidence"             "Channel.Q.Value"              "Decoy"                        "Empirical.Quality"           
>  [9] "Evidence"                     "Fragment.Sum"                 "FWHM"                         "Genes"                       
> [13] "Genes.MaxLFQ"                 "Genes.MaxLFQ.Quality"         "Genes.MaxLFQ.Unique"          "Genes.MaxLFQ.Unique.Quality" 
> [17] "Genes.TopN"                   "GG.Q.Value"                   "Global.Peptidoform.Q.Value"   "Global.PG.Q.Value"           
> [21] "Global.Q.Value"               "iIM"                          "IM"                           "iRT"                         
> [25] "Lib.Peptidoform.Q.Value"      "Lib.PG.Q.Value"               "Lib.PTM.Site.Confidence"      "Lib.Q.Value"                 
> [29] "Mass.Evidence"                "Modification"                 "Modified.Sequence"            "Ms1.Apex.Area"               
> [33] "Ms1.Apex.Mz.Delta"            "Ms1.Area"                     "Ms1.Normalised"               "Ms1.Profile.Corr"            
> [37] "Ms1.Total.Signal.After"       "Ms1.Total.Signal.Before"      "Normalisation.Factor"         "Normalisation.Noise"         
> [41] "Occupied"                     "PEP"                          "Peptidoform.Q.Value"          "PG.MaxLFQ"                   
> [45] "PG.MaxLFQ.Quality"            "PG.PEP"                       "PG.Q.Value"                   "PG.TopN"                     
> [49] "Precursor.Charge"             "Precursor.Id"                 "Precursor.Lib.Index"          "Precursor.Mz"                
> [53] "Precursor.Normalised"         "Precursor.Quantity"           "Predicted.iIM"                "Predicted.IM"                
> [57] "Predicted.iRT"                "Predicted.RT"                 "Probability"                  "Protein"                     
> [61] "Protein.Group"                "Protein.Ids"                  "Protein.Index.In.Group"       "Protein.Names"               
> [65] "Protein.Q.Value"              "Protein.Sites"                "Proteotypic"                  "PTM.Site.Confidence"         
> [69] "Q.Value"                      "Quantity.Quality"             "Residue"                      "RT"                          
> [73] "RT.Start"                     "RT.Stop"                      "Run"                          "Run.Index"                   
> [77] "Sequence"                     "Site"                         "Site.Occupancy.Probabilities" "Stripped.Sequence"           
> [81] "Translated.Q.Value"

8. After having the correct quantity do I just need to `pivot_wider()` the matrix in order to have samples in column and phosphosites in rows ? Or do I need to do extra stuff ?

Sorry for the long post and thank you for your help.

KR

**Comment by vdemichev** [comment]
Hi KR,

> not necessarly only rely on the report.phosphosites_90.tsv or report.phosphosites_99.tsv output

Indeed, for this reason DIA-NN generates site_report.parquet since version 2.2.

> But then what about the quantity that we need to choose

Up to you: can use the fragment sum or any of the quantities in the main report, including MS1 signals. Easiest: use Fragment.Sum and once you have a good 'biological' readout for your experiment, see if changing the quantity used brings any benefits.

> After having the correct quantity do I just need to pivot_wider() the matrix in order to have samples in column and phosphosites in rows ? Or do I need to do extra stuff ?

I assume this is a basic task in R? Best to just have AI agent do all this automatically. Any modern model can one-shot the whole task.

Best,
Vadim

**Reply by KR-15** [reply]
Hi Vadim,

Thank you for your reply.

> I assume this is a basic task in R? Best to just have AI agent do all this automatically. Any modern model can one-shot the whole task.

Yes I know that, but I was just asking if there were any extra steps I might have overlooked - things an AI agent couldn't guess for me.

Best,

KR
